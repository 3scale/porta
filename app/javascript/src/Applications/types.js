// @flow

export type ServicePlan = {
  id: number,
  name: string,
  issuer_id: number, // TODO: change to productId in app/helpers/buyers/applications_helper.rb:14
  default: boolean
}

export type ApplicationPlan = {
  id: number,
  name: string,
  issuer_id: number // TODO: change to productId in app/helpers/buyers/applications_helper.rb:14
}

export type Product = {
  id: number,
  name: string,
  appPlans: ApplicationPlan[],
  servicePlans: ServicePlan[],
  defaultServicePlan: ServicePlan | null
}

export type Buyer = {
  id: string,
  name: string,
  contractedProducts: Product[],
  // servicePlans: ServicePlan[], Not needed?
  createApplicationPath: string,
}
