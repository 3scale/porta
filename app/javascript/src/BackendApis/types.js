// @flow

type ProductLink = {
  name: 'Edit' | 'Overview' | 'Analytics' | 'Methods and Metrics' | 'Mapping Rules',
  path: string
}

export type Backend = {
  id: number,
  name: string,
  systemName: string,
  privateEndpoint: string,
  updatedAt: string,
  links: Array<ProductLink>,
  productsCount: number
}
