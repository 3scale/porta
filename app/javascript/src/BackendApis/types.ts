interface ProductLink {
  name: 'Analytics' | 'Edit' | 'Mapping Rules' | 'Methods and Metrics' | 'Overview';
  path: string;
}

export interface Backend {
  id: number;
  name: string;
  systemName: string;
  privateEndpoint: string;
  updatedAt: string;
  links: ProductLink[];
  productsCount: number;
}
