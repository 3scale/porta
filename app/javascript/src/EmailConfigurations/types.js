// @flow

export type EmailConfiguration = {
  id: number,
  email: string,
  userName: string,
  links: {
    edit: string,
  }
}

export type FormEmailConfiguration = {
  email: string | null,
  userName: string | null,
  password: string | null
}

export type FormErrors = {
  user_name: string[],
  email: string[],
  password: string[]
}
