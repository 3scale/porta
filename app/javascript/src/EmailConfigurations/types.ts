export type EmailConfiguration = {
  id: number,
  email: string,
  userName: string,
  updatedAt: string,
  links: {
    edit: string
  }
}

export type FormEmailConfiguration = {
  id?: number // Means it's updating not creating,
  email: string | null,
  userName: string | null,
  password: string | null
}

export type FormErrors = {
  // eslint-disable-next-line camelcase
  user_name?: string[],
  email?: string[],
  password?: string[]
}
