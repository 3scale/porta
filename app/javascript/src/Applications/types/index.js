// @flow

export type ServicePlan = {
  id: number,
  name: string,
  default: boolean
}

export type ApplicationPlan = {
  id: number,
  name: string,
  serviceName: string,
  contractedServicePlan: ?ServicePlan,
  servicePlans: ?ServicePlan[]
}

export type UserDefinedField = {
  name: string,
  label: string,
  hidden: boolean,
  required: boolean
}
