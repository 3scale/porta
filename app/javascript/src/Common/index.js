// @flow

export * from './components/SelectModal'
