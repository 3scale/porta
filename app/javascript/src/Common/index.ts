export { CompactListCard } from './components/CompactListCard'
export { FancySelect } from './components/FancySelect'
export { MicroPagination } from './components/MicroPagination'
export { NoMatchFound } from './components/NoMatchFound'
export { TableModal } from './components/TableModal'
export { ToolbarSearch } from './components/ToolbarSearch'
export { Select } from './components/Select'
export { SelectWithModal } from './components/SelectWithModal'
export { Spinner } from './components/Spinner'
export { SystemNamePopover, SystemNamePopoverWrapper } from './components/SystemNamePopover'
export { UserDefinedField } from './components/UserDefinedField'
