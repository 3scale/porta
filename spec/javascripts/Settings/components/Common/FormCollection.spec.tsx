import { FunctionComponent } from 'react'
import { shallow } from 'enzyme'

import { FormCollection, Props } from 'Settings/components/Common/FormCollection'
import { FieldGroupProps } from 'Settings/types'

it('should render correctly', () => {
  const GoodBand: FunctionComponent<FieldGroupProps> = ({ name }) => <span>{name} rocks!</span>

  const props: Props = {
    legend: 'Some Good Bands',
    ItemComponent: GoodBand,
    collection: [
      { name: 'The Rolling Stones' },
      { name: 'The Brian Jonestown Massacre' },
      { name: 'Radiohead' },
      { name: 'Blur' },
      { name: 'The Clash' }
    ] as FieldGroupProps[]
  }

  const view = shallow(<FormCollection {...props} />)
  expect(view).toMatchSnapshot()
})
