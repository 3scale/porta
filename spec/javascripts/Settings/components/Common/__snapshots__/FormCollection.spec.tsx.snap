// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<FormFieldset
  id="fieldset-SomeGoodBands"
>
  <FormLegend>
    Some Good Bands
  </FormLegend>
  <GoodBand
    key="The Rolling Stones"
    name="The Rolling Stones"
  />
  <GoodBand
    key="The Brian Jonestown Massacre"
    name="The Brian Jonestown Massacre"
  />
  <GoodBand
    key="Radiohead"
    name="Radiohead"
  />
  <GoodBand
    key="Blur"
    name="Blur"
  />
  <GoodBand
    key="The Clash"
    name="The Clash"
  />
</FormFieldset>
`;
